python/dataproc_templates/gcs/gcs_to_gcs.py

```diff
@@ -0,0 +1,197 @@
+            dest=constants.GCS_TO_GCS_INPUT_FORMAT,
+            required=True,
+            help='GCS input file format (one of: avro,parquet,csv,json)',
+            choices=[
+                constants.FORMAT_AVRO,
+                constants.FORMAT_PRQT,
+                constants.FORMAT_CSV,
+                constants.FORMAT_JSON
+            ]
+        )
+        parser.add_argument(
+            f'--{constants.GCS_TO_GCS_TEMP_VIEW_NAME}',
+            dest=constants.GCS_TO_GCS_TEMP_VIEW_NAME,
+            required=False,
+            default="",
+            help='Temp view name for creating a spark sql view on source data. This name has to match with the table name that will be used in the SQL query'
+        )
+        parser.add_argument(
+            f'--{constants.GCS_TO_GCS_SQL_QUERY}',
+            dest=constants.GCS_TO_GCS_SQL_QUERY,
+            required=False,
+            default="",
+            help='SQL query for data transformation. This must use the temp view name as the table to query from.'
+        )
+        
+        parser.add_argument(
+            f'--{constants.GCS_TO_GCS_OUTPUT_PARTITION_COLUMN}',
+            dest=constants.GCS_TO_GCS_OUTPUT_PARTITION_COLUMN,
+            required=False,
+            default="",
+            help='Partition column name to partition the final output in destination bucket'
+        )
+        parser.add_argument(
+            f'--{constants.GCS_TO_GCS_OUTPUT_FORMAT}',
+            dest=constants.GCS_TO_GCS_OUTPUT_FORMAT,
+            required=False,
+            default=constants.FORMAT_PRQT,
+            help=(
+                'Output write format '
+                '(one of: avro,parquet,csv,json)'
+                '(Defaults to parquet)'
+            ),
+            choices=[
+                constants.FORMAT_AVRO,
+                constants.FORMAT_PRQT,
+                constants.FORMAT_CSV,
+                constants.FORMAT_JSON
+            ]
+        )
+        
+        parser.add_argument(
+            f'--{constants.GCS_TO_GCS_OUTPUT_MODE}',
+            dest=constants.GCS_TO_GCS_OUTPUT_MODE,
+            required=False,
+            default=constants.OUTPUT_MODE_APPEND,
+            help=(
+                'Output write mode '
+                '(one of: append,overwrite,ignore,errorifexists) '
+                '(Defaults to append)'
+            ),
+            choices=[
+                constants.OUTPUT_MODE_OVERWRITE,
+                constants.OUTPUT_MODE_APPEND,
+                constants.OUTPUT_MODE_IGNORE,
+                constants.OUTPUT_MODE_ERRORIFEXISTS
+            ]
+        )
+        
+        parser.add_argument(
+            f'--{constants.GCS_TO_GCS_OUTPUT_LOCATION}',
+            dest=constants.GCS_TO_GCS_OUTPUT_LOCATION,
+            required=True,
+            help=(
+                'destination GCS location'
+            )
+        )
+        
+        known_args: argparse.Namespace
+        known_args, _ = parser.parse_known_args(args)
+
```

Comment:
Add the argument validation here:

```
if getattr(known_args, constants.GCS_TO_GCS_SQL_QUERY) and not getattr(known_args, constants.GCS_TO_GCS_TEMP_VIEW_NAME):
            sys.exit('ArgumentParser Error: Temp view name cannot be null if you want to do data transformations with query')
```

Reply:
Nice Suggestion. Added the same to the code. Please review.